runtime/lua/vim/lsp/buf.lua

```diff
@@ -606,23 +602,27 @@ end
 
 --- @param method string
 local function call_hierarchy(method)
-  local params = util.make_position_params()
-  --- @param result lsp.CallHierarchyItem[]?
-  request(ms.textDocument_prepareCallHierarchy, params, function(err, result, ctx)
-    if err then
-      vim.notify(err.message, vim.log.levels.WARN)
-      return
-    end
-    if not result or vim.tbl_isempty(result) then
-      vim.notify('No item resolved', vim.log.levels.WARN)
-      return
-    end
-    local item = pick_call_hierarchy_item(result)
-    if not item then
-      return
+  lsp.buf_request(
```

Comment:
For potential follow up: We'll also have to change this to merge the results - currently you'd get a inputlist dialog per client & handler call.

Reply:
Ouch, yeah that's gotta change.

Reply:
https://github.com/neovim/neovim/pull/30931